presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -335,6 +338,21 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                             PERFORMANCE_WARNING,
                             "COUNT(DISTINCT xxx) can be a very expensive operation when the cardinality is high for xxx. In most scenarios, using approx_distinct instead would be enough"));
                 }
+                if (functionResolution.isApproxDistinctFunction(analysis.getFunctionHandle(node))) {
```

Comment:
WE should do this for approx_set as well.

Reply:
Lets handle approx_set as well in this PR?